From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Huangbin Zhan <zhanhb88@gmail.com>
Date: Wed, 9 Dec 2020 14:18:11 +0800
Subject: [PATCH] libopkg: allow install package from https and ftps

Signed-off-by: Huangbin Zhan <zhanhb88@gmail.com>
---
 libopkg/opkg_download.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/libopkg/opkg_download.c
+++ b/libopkg/opkg_download.c
@@ -321,8 +321,8 @@ int opkg_prepare_url_for_install(const c
 
 	pkg = pkg_new();
 
-	if (str_starts_with(url, "http://")
-	    || str_starts_with(url, "ftp://")) {
+	if (str_starts_with(url, "http://") || str_starts_with(url, "https://")
+	    || str_starts_with(url, "ftp://") || str_starts_with(url, "ftps://")) {
 		char *tmp_file;
 		char *file_basec = xstrdup(url);
 		char *file_base = basename(file_basec);
