From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Huangbin Zhan <zhanhb88@gmail.com>
Date: Sun, 24 Oct 2021 11:25:08 +0800
Subject: [PATCH] libopkg: preserve auto installed when upgrading

Signed-off-by: Huangbin Zhan <zhanhb88@gmail.com>
---
 libopkg/opkg_install.c | 1 +
 1 file changed, 1 insertion(+)

--- a/libopkg/opkg_install.c
+++ b/libopkg/opkg_install.c
@@ -1415,6 +1415,7 @@ int opkg_install_pkg(pkg_t * pkg, int fr
 	pkg->state_flag |= SF_FILELIST_CHANGED;
 
 	if (old_pkg) {
+		pkg->auto_installed = old_pkg->auto_installed;
 		pkg_remove_orphan_dependent(pkg, old_pkg);
 		old_pkg->is_upgrade = 1;
 		pkg->is_upgrade = 1;
