From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Huangbin Zhan <zhanhb88@gmail.com>
Date: Mon, 8 Nov 2021 13:14:04 +0800
Subject: [PATCH] libopkg: file_util strcpy instead of strncat

Signed-off-by: Huangbin Zhan <zhanhb88@gmail.com>
---
 libopkg/file_util.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

--- a/libopkg/file_util.c
+++ b/libopkg/file_util.c
@@ -59,24 +59,24 @@ int file_is_dir(const char *file_name)
 */
 char *file_read_line_alloc(FILE * fp)
 {
-	size_t buf_len, line_size;
+	size_t buf_len, line_size = 0;
 	char buf[BUFSIZ];
 	char *line = NULL;
 	int got_nl = 0;
+	buf[BUFSIZ - 1] = '\0';
 
 	while (fgets(buf, BUFSIZ, fp)) {
 		buf_len = strlen(buf);
 		if (buf_len > 0 && buf[buf_len - 1] == '\n') {
-			buf_len--;
-			buf[buf_len] = '\0';
+			buf[--buf_len] = '\0';
 			got_nl = 1;
 		}
 		if (line) {
+			line = xrealloc(line, line_size + buf_len + 1);
+			strcpy(&line[line_size], buf);
 			line_size += buf_len;
-			line = xrealloc(line, line_size + 1);
-			strncat(line, buf, line_size);
 		} else {
-			line_size = buf_len + 1;
+			line_size = buf_len;
 			line = xstrdup(buf);
 		}
 		if (got_nl)
